### DISCUSSION TITLE
Is sqlite auto increment supported?

### DISCUSSION THREAD

**Discussion by fairking** [fluentmigrator/fluentmigrator#1606] [Q&A]
**Describe the question**
Is sqlite auto increment supported?
As far as I understand it is not. I tried the following code but it seems it's not working:
``` c#
Create.Table("Customers")
    .WithColumn("Id").AsInt32().Identity().PrimaryKey()
    .WithColumn("Name").AsString(50).NotNullable()
    ;
```

**Documentation Pages You've Read So Far**
https://fluentmigrator.github.io/articles/extensions/sql-server-extensions.html#add-identity-column-with-seed-and-increment

**Expected benefit**
To apply autoincrement primary keys to sqlite db.

**Information (please complete the following information):**
 - OS: [ Windows 10 ]
 - Platform [ .NET Core 6 ]
 - FluentMigrator version [ fluentmigrator\3.3.2\ ]
 - FluentMigrator runner [ fluentmigrator.runner.sqlite\3.3.2\ ]
 - Database Management System [ SQLite ]
 - Database Management System Version [ system.data.sqlite.core\1.0.115.5\ ]

**Additional context**
https://sqlite.org/autoinc.html

**Comment by jzabroski** [comment]
@mattbrailsford can you answer?

> **Documentation Pages You've Read So Far** https://fluentmigrator.github.io/articles/extensions/sql-server-extensions.html#add-identity-column-with-seed-and-increment

That is for SQL Server.  We also support Postgres, but I dont think we support SQLite.

**Reply by fairking** [reply]
Thanks John for opening this discussion. 👍 

I personally don't use autoincrement, but in one of side projects it is used. At the moment as a workaround we have to create tables by providing raw sql.

**Reply by jzabroski** [reply]
@fairking - @mattbrailsford added support in https://github.com/fluentmigrator/fluentmigrator/pull/1583 for more robust identity.
The workaround on Fm 3.3.2 is to use DbType.Int32 for your column type.
If you would like to submit an FAQ doc for SQLite that would be great.
